app/controllers/settings/base_certificate_controller.rb

```diff
@@ -95,12 +96,30 @@ def certificate_holder_update_params
           "#{self.class.name}#certificate_holder_update_params is an abstract method."
   end
 
+  def passed_certificate
+    # Storing the cert in the variable is required, as uploaded files (in RSpec
+    # tests at least) will not be available for multiple calls:
+    #
+    # > passed_certificate.present? => true
+    #
+    # > passed_certificate.present?  => false
+    if @passed_cert.present?
+      @passed_cert
```

Comment:
this statement is a bit useless here, as it's already done outside the condition, maybe early return here?

or simply:

```ruby
def passed_certificate
  @passed_cert ||= if certificate_holder_params[:certificate].present? ||
    certificate_holder_params[:current_cert].present?
    return Certificate.find_or_initialize_by(
      certificate: Certificate.get_certificate_text(certificate_holder_params)
    )
  end
end
```

Reply:
Nice, I didn't know about the `||=` operator in ruby - looks quite handy.

I think I only have to remove the return, otherwise it won't assign anything to `@passed_cert`:

```ruby
[3] pry(main)> a ||= if 0 == 0
[3] pry(main)*   return 1
[3] pry(main)* end
SyntaxError: (eval):4: void value expression
```

Reply:
oops, yes, the `return` was wrong ;) yeah, the `||=` is often used in ruby to memoize variables